src/pages/index.js

```diff
@@ -33,10 +46,47 @@ constants.profileEditButton.addEventListener("click", () => {
   profileFormValidator.resetValidation();
 });
 
-function handleProfileFormSubmit({ name, description }) {
-  user.setUserInfo({ name, description });
+function handleProfileFormSubmit({ name, about }) {
+  profileEditModal.setLoading(true);
+  api
+    .updateUserProfile({ name, about })
+    .then(() => {
+      user.setUserInfo({ name, about });
+    })
+    .finally(() => {
+      profileEditModal.setLoading(false);
+      profileEditModal.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again. (fix that here and in all the other places)

Reply:
Needs correction: please move `profileEditModal.close();` line to `then` block, which is located on line 53